src/dropdown/dropdown.ts

```diff
@@ -242,7 +252,63 @@ export class NgbDropdown implements OnInit, OnDestroy {
 
   private _positionMenu() {
     if (this.isOpen() && this._menu) {
-      this._menu.position(this._anchor.anchorEl, this.placement);
+      this._position(this._anchor.anchorEl, this.placement, this.container);
+    }
+  }
+
+  private _position(triggerEl, placement, container) {
+    this._applyPlacement(
+        positionElements(
+            triggerEl, this._bodyContainer || this._menuElement.nativeElement, placement, container === 'body'));
+  }
+
+  private _applyContainer(container: null | 'body' = null) {
+    const renderer = this._renderer;
+    const body = this._document.body;
+    const dropdownElement = this._elementRef.nativeElement;
+    const dropdownMenuElement = this._menuElement.nativeElement;
+
+    if (container === 'body') {
+      const bodyContainer = this._bodyContainer = this._bodyContainer || renderer.createElement('div');
+
+      // Override some styles to have the positionning working
+      renderer.setStyle(this._bodyContainer, 'position', 'absolute');
+      renderer.setStyle(dropdownMenuElement, 'position', 'static');
+
+      renderer.appendChild(bodyContainer, dropdownMenuElement);
+      renderer.appendChild(body, bodyContainer);
+    } else {
+      renderer.appendChild(dropdownElement, dropdownMenuElement);
+      renderer.removeStyle(dropdownMenuElement, 'position');
+      if (this._bodyContainer) {
+        renderer.removeChild(body, this._bodyContainer);
+        this._bodyContainer = null;
+      }
+    }
+  }
+
+  private _applyPlacement(_placement: Placement) {
+    const renderer = this._renderer;
+    const dropdownElement = this._elementRef.nativeElement;
+
+    // remove the current placement classes
+    renderer.removeClass(dropdownElement, 'dropup');
+    renderer.removeClass(dropdownElement, 'dropdown');
+    this.placement = _placement;
+    this._menu.setPlacement(_placement);
+
+    /*
+     * apply the new placement
+     * in case of top use up-arrow or down-arrow otherwise
+     */
+    const dropdownClass = _placement.search('^top') !== -1 ? 'dropup' : 'dropdown';
+    renderer.addClass(dropdownElement, dropdownClass);
+
+    const bodyContainer = this._bodyContainer;
+    if (bodyContainer) {
```

Comment:
So in case of a valid `bodyContainer` you'll add placement and dropdown classes on BOTH `bodyContainer` and `dropdownElement`? don't get why

Reply:
It's because of the arrow (and, by extension, on the theme customization that can be applied). This is what is get if I just set it in the body container only :

![image](https://user-images.githubusercontent.com/1152930/49282508-a1643600-f48f-11e8-9cac-986780500f29.png)